### PR TITLE
CEL Library: errors improvement

### PR DESCRIPTION
**Pull request by jiahuif**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
This PR utilizes go error wrapping to pass on errors so that the caller can get the root cause without resorting to error message matching. Note that an error message that comes from third-party libraries, including cel-go, may not be reliable across dependency updates.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by jiahuif** [comment]
/assign @cici37 @alexzielenski
Can either of you take a look? We can approve this PR locally if it looks good to you.

**Review by cici37 — COMMENTED** [review]
I am ok with having more detailed error types but wanna point out that the existing error message matching does not depend on third party error message hence it should not have the concern of third party library change :)

File: staging/src/k8s.io/apiserver/pkg/admission/plugin/cel/compile.go [hunk 1]
```diff
@@ -163,11 +163,12 @@ type variableDeclEnvs map[OptionalVariableDeclarations]*environment.EnvSet
 // CompileCELExpression returns a compiled CEL expression.
 // perCallLimit was added for testing purpose only. Callers should always use const PerCallLimit from k8s.io/apiserver/pkg/apis/cel/config.go as input.
 func (c compiler) CompileCELExpression(expressionAccessor ExpressionAccessor, options OptionalVariableDeclarations, envType environment.Type) CompilationResult {
-	resultError := func(errorString string, errType apiservercel.ErrorType) CompilationResult {
+	resultError := func(errorString string, errType apiservercel.ErrorType, errors ...error) CompilationResult {
 		return CompilationResult{
 			Error: &apiservercel.Error{
 				Type:   errType,
 				Detail: errorString,
+				Errors: errors,
```
[the remark this answers is not in the document]
Comment (jiahuif) [reply staging/src/k8s.io/apiserver/pkg/admission/plugin/cel/compile.go]: Nope. Changed to warp zero or one error. PTAL.

**Comment by cici37** [comment]
hello @jiahuif, just to confirm are you still planning to work on this PR? Thanks :)

**Comment by cici37** [comment]
/close
The commits are going to be merged in https://github.com/kubernetes/kubernetes/pull/126123

**Pull request closed without merging**